### PR TITLE
fix(core/menu): fix menu avatar as trigger

### PR DESCRIPTION
**Pull request by matthiashader**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`yarn build`) was run locally and any changes were pushed
- [x] Unit tests (`yarn test`) were run locally and passed
- [x] Visual Regression Tests (`yarn visual-regression`) were run locally and passed
- [x] Linting (`npm lint`) was run locally and passed

## Pull request type

Please check the type of change your PR introduces:

- [x] Bug fix
- [ ] Feature
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

- ix-menu-avatar-item does not work as dropdown trigger

GitHub Issue Number: #895

## What is the new behavior?

- ix-menu detects that the avatar-item contains a ix-dropdown and now has the same behaviour

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by nuke-ellington — CHANGES_REQUESTED** [review]
Test cases missing

File: packages/core/src/components/dropdown/dropdown.tsx [hunk 1]
```diff
@@ -357,7 +357,7 @@ export class Dropdown {
 
     const { defaultPrevented } = this.showChanged.emit(!this.show);
 
-    if (!defaultPrevented) {
+    if (!defaultPrevented && !this.show) {
```
Comment (danielleroux) [inline packages/core/src/components/dropdown/dropdown.tsx:360]: In this case toggle can only be prevented if the show is not already shown. This makes for a function "toggle" which change between show=true and show=false no sense.

**Review by danielleroux — CHANGES_REQUESTED** [review]

File: packages/core/src/components/menu-avatar/menu-avatar.tsx [hunk 2]
```diff
@@ -86,6 +87,26 @@ export class MenuAvatar {
...
+    const elementsWithIds = this.hostElement.querySelectorAll('[id]');
+    const elementIds = Array.from(elementsWithIds).map((e) => e.id);
+
+    for (let i of elementIds) {
+      const element = document.querySelector('[' + i.toString() + ']');
+      if (element) {
+        (element as HTMLElement).style.zIndex = `var(--theme-z-index-sticky)`;
+      }
+    }
+  }
```
Comment (danielleroux) [inline packages/core/src/components/menu-avatar/menu-avatar.tsx:100]: What do you want to achieve here?

**Review by danielleroux — COMMENTED** [review]
Can you also explain why the images are updated? Cannot find a visual indicator for a change.

**Comment by jul-lam** [comment]
> Can you also explain why the images are updated? Cannot find a visual indicator for a change.

The only plausible explanation appears to be the removal of the line `z-index: var(--theme-z-index-fixed);` within the application header. This modification was essential to address the z-index issue associated with the nested avatar dropdown.

**Comment by danielleroux** [comment]
#1060

**Pull request closed without merging**